benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/strategy/selection/CompleteSelectionStrategy.java

```diff
@@ -0,0 +1,80 @@
+package cloud.benchflow.testmanager.strategy.selection;
+
+import cloud.benchflow.dsl.BenchFlowDSL;
+import cloud.benchflow.dsl.definition.BenchFlowExperiment;
+import cloud.benchflow.dsl.definition.errorhandling.BenchFlowDeserializationException;
+import cloud.benchflow.testmanager.BenchFlowTestManagerApplication;
+import cloud.benchflow.testmanager.exceptions.BenchFlowTestIDDoesNotExistException;
+import cloud.benchflow.testmanager.services.external.MinioService;
+import cloud.benchflow.testmanager.services.internal.dao.BenchFlowTestModelDAO;
+import cloud.benchflow.testmanager.services.internal.dao.ExplorationModelDAO;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@usi.ch)
+ *         created on 2017-04-20
+ */
+public class CompleteSelectionStrategy implements ExperimentSelectionStrategy {
+
+    private static Logger logger = LoggerFactory.getLogger(CompleteSelectionStrategy.class.getSimpleName());
+
+    private final MinioService minioService;
+    private final ExplorationModelDAO explorationModelDAO;
+    private final BenchFlowTestModelDAO testModelDAO;
+
+    public CompleteSelectionStrategy() {
+
+        this.minioService = BenchFlowTestManagerApplication.getMinioService();
+        this.explorationModelDAO = BenchFlowTestManagerApplication.getExplorationModelDAO();
+        this.testModelDAO = BenchFlowTestManagerApplication.getTestModelDAO();
+    }
+
+    // only used for testing
+    public CompleteSelectionStrategy(MinioService minioService, ExplorationModelDAO explorationModelDAO, BenchFlowTestModelDAO testModelDAO) {
+        this.minioService = minioService;
+        this.explorationModelDAO = explorationModelDAO;
+        this.testModelDAO = testModelDAO;
+    }
+
+    @Override
+    public String selectNextExperiment(String testID) {
+
+        logger.info("selectNextExperiment: " + testID);
+
+        try {
+
+            String testDefinitionYamlString = IOUtils.toString(minioService.getTestDefinition(testID), StandardCharsets.UTF_8);
+
+            // get exploration space
+            // TODO - generalize this to complete search space
```

Comment:
Lets do the DSL change in another branch, and reference to the new features here. As it is now it cannot be considered complete.